daliuge-engine/dlg/drop.py

```diff
@@ -2353,6 +2373,30 @@ def streamingInputs(self) -> List[DataDROP]:
         """
         return list(self._streamingInputs.values())
 
+    def _generateNamedInputs(self):
+        """
+        Generates a named mapping of input data drops. Can only be called during run().
+        """
+        named_inputs: OrderedDict[str, DataDROP] = OrderedDict()
+        if 'inputs' in self.parameters and isinstance(self.parameters['inputs'][0], dict):
+            for i in range(len(self._inputs)):
+                key = list(self.parameters['inputs'][i].values())[0]
+                value = self._inputs[list(self.parameters['inputs'][i].keys())[0]]
+                named_inputs[key] = value
+        return named_inputs
```

Comment:
Could possibly use this in pyfunc but there's some additional splitting of `posargs` from `kwargs` needed that is questionable whether useful for other components